sockeye/inference.py

```diff
@@ -1326,13 +1362,32 @@ def _beam_search(self,
 
             # (7) determine which hypotheses in the beam are now finished
             finished = ((best_word_indices == C.PAD_ID) + (best_word_indices == self.vocab_target[C.EOS_SYMBOL]))
-            if mx.nd.sum(finished).asscalar() == self.batch_size * self.beam_size:  # all finished
-                break
+
+#            self._print_beam(sequences, scores_accumulated, finished, inactive, t)
+
+            if self.beam_search_stop == 'first' and self.batch_size == 1:
+                # TODO: doesn't work for batching
```

Comment:
what doesn't work for batching? 'first' top and batch_size > 1? But then this line is unreachable based on 1368, right?

Reply:
For batching, we need to check if there is a sentence completed in every sentence in the batch. The whole concept of early stopping is a little funny with batch processing, so I didn't bother doing this check. We could remove early stopping if you'd like (though I don't think it hurts).